src/Bicep.Core/TypeSystem/DeclaredTypeManager.cs

```diff
@@ -89,7 +92,7 @@ public DeclaredTypeManager(IResourceTypeProvider resourceTypeProvider, TypeManag
                     return GetArrayAccessType(arrayAccess);
 
                 case VariableDeclarationSyntax variable:
-                    return new DeclaredTypeAssignment(this.typeManager.GetTypeInfo(variable), variable);
+                    return new DeclaredTypeAssignment(LanguageConstants.Any, variable);
```

Comment:
Why we're changing this to Any type?

Reply:
Yeah we'll have to revisit this when we add better completions for property access on array or object literals, but that can wait.